setup.py

```diff
@@ -5,5 +5,28 @@
 """
 from setuptools import setup
 
+# TODO (pypa/setuptools#3221): Migrate this to `[project.optional-dependencies]` in
+# `pyproject.toml`, once optional dependencies defined there can reference one another
+extras_require = dict()
+extras_require["jax"] = [
+    "jax[cpu]<0.3.5; platform_system!='Windows'",
+]
+extras_require["zoo"] = [
+    "tqdm>=4.0",
+    "requests>=2.0",
+] + extras_require["jax"]
+extras_require["pls_calib"] = [
+    "GPy",
+    # GPy can't be imported without matplotlib.
+    # This is and should not be a ProbNum dependency.
+    "matplotlib",
+]
+extras_require["full"] = (
+    extras_require["jax"] + extras_require["zoo"] + extras_require["pls_calib"]
+)
```

Comment:
This is unfortunate, but I still think it is better to add this here and get rid of `setup.cfg` now.
We need `setup.py` anyway and it is easier to migrate this to `pyproject.toml` once this is resolved than repeating this PR then.

Reply:
Agreed.